### ISSUE TITLE
[onert] Apply softmax to CategoricalCrossEntropy automatically

### ISSUE THREAD

**Issue by ragmani** [Samsung/ONE#14028]
## What

Let's apply softmax(nomalization) automatically when trying to use CategoricalCrossEntropy loss.

## Why
There is no guarantee a pre-trained model has a softmax when trying to apply CategoricalCrossEntropy loss to the model.

## To do
- Investigate how softmax exists when a circle model is created from Pytorch and Tensorflow.
  - Pytorch : Circle models don't have softmax. Pytorch does not add softmax to models even when training with CategoricalCrossEntropy.
  - Tensorflow : Circle models may have softmax if users add softmax. Tensorflow trains well with CategoricalCrossEntropy by applying softmax only once even if there is softmax in the model. 
- Apply softmax to CategoricalCrossEntropy automatically

---

To apply normalization(softmax) automatically to categorical cross entropy, we need to consider that sum of labels is not 1.
That consideration will be deal with in another issue later. So, I'm closing this issue since I have completed all other required tasks.

_Originally posted by @ragmani in https://github.com/Samsung/ONE/issues/13736#issuecomment-2342763173_

**Comment by ragmani** [comment]
4 cases of circle models pre-trained with `CategoricalCrossEntropy` can be created:

1. A model **with** softmax, trained by executing softmax **once** for each step (on tensorflow).
2. A model **without** softmax, trained by executing softmax **once** for each step(on tensorflow and pytorch).
3. A model **with** softmax, trained by executing softmax **twice** for each step(on pytorch).
4. A model **without** softmax, trained **without executing softmax**(on tensorflow and pytorch).

**Comment by ragmani** [comment]
Done.
